server/src/dynamodb.ts

```diff
@@ -0,0 +1,128 @@
+import * as AWS from 'aws-sdk';
+import * as uuid from 'uuid';
+import {
+  Result,
+  DatabaseType,
+  Database,
+  DatabaseModel,
+} from '../../frontend/src/types';
+import { dynamoConfigOptions, tableName } from './config';
+
+export class Dynamo implements Database {
+  public type = DatabaseType.Dynamo;
+  dynamoDb: AWS.DynamoDB.DocumentClient;
+
+  constructor() {
+    this.dynamoDb = new AWS.DynamoDB.DocumentClient(dynamoConfigOptions);
+  }
+
+  public async init() {
+    if (process.env.CREATE_DYNAMO_TABLES) {
+      const params = {
+        TableName: tableName,
+        KeySchema: [{ AttributeName: 'id', KeyType: 'HASH' }],
+        AttributeDefinitions: [{ AttributeName: 'id', AttributeType: 'S' }],
+        ProvisionedThroughput: {
+          ReadCapacityUnits: 5,
+          WriteCapacityUnits: 5,
+        },
+      };
+
+      try {
+        const dynamoClient = new AWS.DynamoDB(dynamoConfigOptions);
+        console.log('dynamo config: ', dynamoConfigOptions);
+        await dynamoClient.createTable(params).promise();
+        const data = await dynamoClient.listTables().promise();
+        console.log('Created table.', JSON.stringify(data, null, 2));
+      } catch (err) {
+        if (err.message != 'Cannot create preexisting table')
+          console.log(`Unable to create table:`, err.message);
+      }
+    }
+  }
+
+  public async storeProject(projectInfo: Result): Promise<string> {
```

Comment:
similarly to my comment in the `Disk` type. I would expected the parameter type to be `DatabaseModel`. The mapping logic appears identical which to me suggests it's a good candidate to be handled outside this function